src/halfutils.c

```diff
@@ -244,6 +363,15 @@ HalfvecInit(void)
 		HalfvecL2SquaredDistance = HalfvecL2SquaredDistanceF16c;
 		HalfvecInnerProduct = HalfvecInnerProductF16c;
 		HalfvecCosineSimilarity = HalfvecCosineSimilarityF16c;
+
+#ifdef HAVE_AVX512FP16
+		if (SupportsAvx512Fp16())
+		{
+			HalfvecL2SquaredDistance = HalfvecL2SquaredDistanceAvx512Fp16;
+			HalfvecInnerProduct = HalfvecInnerProductAvx512Fp16;
+			HalfvecCosineSimilarity = HalfvecCosineSimilarityAvx512Fp16;
+		}
+#endif
```

Comment:
nitpick:  This might not need to be nested in the `HALFVEC_DISPATCH` block.

Reply:
You're right. Currently, it is taking advantage of the OSXSAVE check included with the other features, but I'll separate that.